nengo_fpga/networks/fpga_pes_ensemble_network.py

```diff
@@ -428,8 +440,61 @@ def connect(self):
         connect_thread = threading.Thread(target=self.connect_thread_func, args=())
         connect_thread.start()
 
+        logger.info("<%s> Open UDP connection", fpga_config.get(self.fpga_name, "ip"))
+        # Create a UDP socket to communicate with the board
+        self.udp_socket = socket.socket(socket.AF_INET, socket.SOCK_DGRAM)
+        self.udp_socket.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        self.udp_socket.bind((fpga_config.get("host", "ip"), self.udp_port))
+
+        # # Set the socket timeout to the connection timeout and wait for the
+        # # board to connect to the PC.
+        # self.udp_socket.settimeout(self.connect_timeout)
+        # while True:
+        #     try:
+        #         self.udp_socket.recv_into(self.recv_buffer.data)
+        #         # Connection packet has a t of 0
+        #         if self.recv_buffer[0] <= 0.0:
+        #             break
+        #     except socket.timeout:
+        #         self.close()
+        #     raise RuntimeError("Did not receive connection from board within "
+        #                        + "specified timeout (%fs)." % self.connect_timeout)
+        # if self.recv_buffer[0] < 0.0:
+        #     self.close()
+        #     raise RuntimeError("Simulation terminated by FPGA board.")
+
+        # Connection with the board established. Set the socket timeout to
+        # recv_timeout.
+        self.udp_socket.settimeout(self.recv_timeout)
+
+        # Wait for the connection packet from the board. Note that the
+        # implementation above (setting the socket timeout to ``connect_timeout``)
+        # is the "proper" implementation. Below is a workaround to allow the SSH
+        # connection thread to terminate the connection process (by closing the
+        # udp socket, so an exception is thrown).
+        max_attempts = int(self.connect_timeout / self.recv_timeout)
+        for conn_attempts in range(max_attempts):
+            try:
+                self.udp_socket.recv_into(self.recv_buffer)
+                if self.recv_buffer[0] <= 0.0:
```

Comment:
Do we want to add a comment somewhere in here stating that we are looking for 0 on success or -1 on failure or something?